client/my-sites/checkout/src/components/checkout-sidebar-plan-upsell/index.tsx

```diff
@@ -16,10 +16,76 @@ import {
 	getItemVariantCompareToPrice,
 	getItemVariantDiscountPercentage,
 } from '../item-variation-picker/util';
+import type { WPCOMProductVariant } from '../item-variation-picker';
 import './style.scss';
 
 const debug = debugFactory( 'calypso:checkout-sidebar-plan-upsell' );
 
+function getUpsellVariant( currentVariant: WPCOMProductVariant, variants: WPCOMProductVariant[] ) {
+	// Return the binennial plan if the plan in cart is a monthly or annual plan
+	if (
+		currentVariant.productBillingTermInMonths === 1 ||
+		currentVariant.productBillingTermInMonths === 12
+	) {
+		return variants?.find( ( product ) => product.termIntervalInMonths === 24 );
+	}
+
+	// Return the triennial plan if the plan in cart is a binennial plan
+	if ( currentVariant.productBillingTermInMonths === 24 ) {
```

Comment:
This logic could be simplified to just look for the next 12 month billing term based on the `currentVariants` billing term. I suppose it would lose granularity, but it would dedupe this function quite a bit if we added more and more upgrade options.

An alternative would be to rethink this sidebar upsell component and instead just show all of our upgrade possibilities in a selector. If the customer is looking to buy in bulk then we may want to 'save them a click' so that they can just directly select the term they want, rather than clicking through each successive upgrade.

Reply:
Currently, there are only two possible upgrade options, and explicitly stating the upgrade paths makes the function more readable, IMO. Maybe we can consider the refactor whenever we add another term?

Reply:
Works for me! 👍